### PR TITLE
Update cs0236.md

### LINKED ISSUE #24865: how on earth is this meant to help me fix the issue
[Enter feedback here]

it just shows me how to make the error not fix it
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 19167a3c-283f-e0c1-d1cb-3b773954779b
* Version Independent ID: 360140ec-e61d-3117-e7a7-92473887f7b6
* Content: [Compiler Error CS0236](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0236)
* Content Source: [docs/csharp/misc/cs0236.md](https://github.com/dotnet/docs/blob/main/docs/csharp/misc/cs0236.md)
* Product: **dotnet-csharp**
* Technology: **csharp-diagnostics**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #24865

### CODE REVIEW

File: docs/csharp/misc/cs0236.md [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
-   MyClass()  
-   {  
-      k = i;  
-   }  
-  
-   public static void Main()  
-   {  
-   }  
-}  
+A field initializer cannot reference the non-static field, method, or property 'name'
```
Comment (IEvangelist) [inline docs/csharp/misc/cs0236.md:12]:
```suggestion
A field initializer cannot reference the non-static field, method, or property 'name'.
```

File: docs/csharp/misc/cs0236.md [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
+{
+    public int i = 5;
+
+    // To fix the error, remove "= i", and uncomment the line in constructor.
+    public int j = i;  // CS0236
+
+    public MyClass()
+    {
+        // Uncomment the following.
+        //k = i;
```
Comment (IEvangelist) [inline docs/csharp/misc/cs0236.md:35]:
```suggestion
        //j = i;
```

You have `k`, but that isn't defined:
> error CS0103: The name 'k' does not exist in the current context

**Review by IEvangelist — CHANGES_REQUESTED** [review]

File: docs/csharp/misc/cs0236.md (hunk 1 above)
Comment (Youssef1313) [reply docs/csharp/misc/cs0236.md]: @IEvangelist In all compiler message articles, this first paragraph is an exact match of the error message of the compiler (with the same punctuation). But I'm going to apply the change.

File: docs/csharp/misc/cs0236.md (hunk 2 above)
Comment (Youssef1313) [reply docs/csharp/misc/cs0236.md]: Good catch. Thanks!

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**